apps/app/src/features/rate-limiter/middleware/consume-points.ts

```diff
@@ -0,0 +1,27 @@
+import { type RateLimiterMongo, type RateLimiterRes } from 'rate-limiter-flexible';
+
+import { DEFAULT_MAX_REQUESTS, type IApiRateLimitConfig } from '../config';
+
+export const consumePoints = async(
+    rateLimiter: RateLimiterMongo, method: string, key: string | null, customizedConfig?: IApiRateLimitConfig, maxRequestsMultiplier?: number,
+): Promise<RateLimiterRes | undefined> => {
+  if (key == null) {
+    return;
+  }
+
+  let maxRequests = DEFAULT_MAX_REQUESTS;
+
+  // use customizedConfig
+  if (customizedConfig != null && (customizedConfig.method.includes(method) || customizedConfig.method === 'ALL')) {
+    maxRequests = customizedConfig.maxRequests;
+  }
+
+  // multiply
+  if (maxRequestsMultiplier != null) {
+    maxRequests *= maxRequestsMultiplier;
+  }
+
+  rateLimiter.points = maxRequests;
+  const rateLimiterRes = await rateLimiter.consume(key, 1);
```

Comment:
機能改修はここのみ

# Before
```typescript
// because the maximum request is reduced by 1 if it is divisible by	
// https://github.com/weseek/growi/pull/6225	
const consumePoints = (POINTS_THRESHOLD + 0.0001) / maxRequests;	
await rateLimiter.consume(key, consumePoints);
```
maxRequest: 1 の場合 2 回目の request で rate limit error べきだが、  (100 + 0.0001) / 1 =  100.0001 になるので1回目の request で rate limit error になる。


# Before (この PR の当初の改修)
```typescript
const consumePoints = POINTS_THRESHOLD / maxRequests;
const rateLimiterRes = await rateLimiter.consume(key, consumePoints);
```

maxRequest: 500 の場合 100 / 500 = 0.2 になる。0.2 * 500 で 100 (閾値) に達するので良いと思っていた。1回の request で消費ポイントが 0.2。2回目の request で消費ポイントが 0.4。3回目の request で 消費ポイントが 0.6000000000000001 になり、期待する通りに動かなかったためこれも却下になった。

## 参考
https://qiita.com/kihara-takahiro/items/4518902225304f4e15b6#%E5%B0%8F%E6%95%B0%E7%82%B9%E3%81%AE%E5%8F%96%E3%82%8A%E6%89%B1%E3%81%84%E6%B3%A8%E6%84%8F


# After (最終的に)
```typescript
rateLimiter.points = maxRequests;
const rateLimiterRes = await rateLimiter.consume(key, 1);
```
RateLimiterMongo 初期化時には point の閾値を設定せず、request 毎に更新するように変更し、point 消費は 1 (整数) で固定するように変更した